### ISSUE TITLE
[BUG?] One page App with tables on two pages at the same level gets them mixed

### ISSUE THREAD

**Issue by noncom** [reagent-project/reagent#185]
I am developing a simple single page app with a nav menu which allows moving between the pages. There are two pages which have tables at the same level (so they have the same reagent id). When I switch from one page to another, weird things happen:

* the "additional" rows of the table with more rows don't get any styling
* moving to the table with less rows throws an exception that there is no such element, apparently referring to the fact that there is no data for the "additional" rows that it wants to draw, causing the app to freeze until I reload

Ideally these should be two separate tables, but they get mixed and make the application unuseable. I have fixed this by embedding one of them in an additional `[:div]` so that it has a different reagent id, but that is not a real solution.

The active page area redraw is triggered by a the reagent `atom`, which tracks the current page id and changes in response to chosing the page from the nav menu.

Looks like a bug?

**Comment by GetContented** [comment]
What happens when you give it some meta data ^{:key "a unique id"} where "a unique id" is some string that is different for each?

**Comment by noncom** [comment]
@GetContented Okay, let me try, but I'm not sure where should I add such metadata? I am following this pattern of creating a table: https://github.com/reagent-project/reagent-cookbook/tree/master/recipes/data-tables

**Comment by GetContented** [comment]
Oh sorry I didn't realise you were using a jquery plugin. I thought you were just using a normal table.

If you're following that recipe, they use a jquery call on document read, which will attach the jquery functionality directly after the page loads, but depending on how your "page" selection functionality works, you might not actually have the second page in the dom "on load", which would mean it'd potentially get confused. I'm not entirely sure how data-tables works, so I'd advise you to put this issue on that recipe, not on reagent proper (ie here)... because it's really not a bug with reagent, it's a problem with your particular instance of implementing the data tables in conjunction with that recipe.

In other words, if you can do their example verbatim and it works, then reagent has no bug. I suggest opening an issue on the recipes repo instead.

**Comment by mike-thompson-day8** [comment]
@noncom  given the explanations provided, can this be closed?

**Comment by noncom** [comment]
Ah, wait a sec, sorry, had a very loaded week. Now I looked into this and I am a bit confused. I did not use jQuery. All I use is reagent + bootstrap. By pointing to the example I meant that my table is generated as shown in Step 3 - by a function that returns a simple vector for building HTML, and I do not know where do I add any meta to it.

There is jQuery added to my deps, but I do not write any code for it or use any HTML components classes besides what is provided by bootstrap.

Sorry if I am being too vague, or something, I am a newb in web and things do not look very clear to me..

Probably I could setup a separate clean project as a testcase for that (I am using a Luminus template), but setting up a clear project for ClojureScript does not seem very straightforward and involves much setup. I could try this on the weekend..

**Comment by mike-thompson-day8** [comment]
There doesn't seem to be a bug, so I'm going to close this.

I'd encourage you to create a minimal Gist or repo showing the problem/question you have, and then raise it in one of the various help forums like:

  - http://clojurians.net/ (#reagent channel)
  - https://groups.google.com/forum/#!forum/clojurescript

**Comment by noncom** [comment]
Okay.
